table/block_based/filter_policy.cc

```diff
@@ -381,9 +469,13 @@ class FastLocalBloomBitsBuilder : public XXPH3FilterBitsBuilder {
 
     // Prime the buffer
     size_t i = 0;
+    bool keep_entries_for_postverify = detect_filter_construct_corruption_;
     for (; i <= kBufferMask && i < num_entries; ++i) {
-      uint64_t h = hash_entries_.front();
-      hash_entries_.pop_front();
+      uint64_t h = hash_entries_info_.entries.front();
+      hash_entries_info_.entries.pop_front();
+      if (keep_entries_for_postverify) {
```

Comment:
This approach of pushing entries back on as we pop them off is not ideal for efficiency (or reliability) either. Conceptually, we are always iterating over the entries but sometimes destroying them as we go and sometimes keeping them. This could be a template function that takes different kind of iterators to use.

Actually, destroying them as we go may not be critical to efficiency (can be tested with filter_bench), so maybe template not needed.

Reply:
Make sense - we can change deque to list structure as well as do not destroy entries in this function. Let's see how the performance change. Fixing it.

Reply:
EDIT: please ignore my prev comment in here - it is incorrect.